File: server/src/learner-activity/core/models/accuracy-rate.model.ts
export class AccuracyRate {
  correct!: number;
  total!: number;
  value!: number;

  constructor({ correct, total }: ConstructorProps) {
    this.correct = Number(correct);
    this.total = Number(total);
    this.value = this.total > 0 ? this.correct / this.total : 0;
  }
}

type ConstructorProps = {
  correct: number;
  total: number;
};

File: client/lib/features/statistics/logic/models/accuracy_rate.model.dart
class AccuracyRate {
  final int correct;
  final int total;
  final double value;

  AccuracyRate({
    required this.correct,
    required this.total,
    required this.value,
  });

  factory AccuracyRate.fromJson(Map<String, dynamic> json) {
    return AccuracyRate(
      correct: json['correct'],
      total: json['total'],
      value: json['value']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'correct': correct,
      'total': total,
      'value': value,
    };
  }
}
